### PR TITLE
docs: document all the captures in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ObserverOfTime**
I used `scheme` as the language marker so that GitHub can highlight the comments properly.

### CODE REVIEW

**Comment by clason** [comment]
Gotta adapt the CI script that scrapes these then, I'm afraid...

**Comment by ObserverOfTime** [comment]
> Gotta adapt the CI script that scrapes these then, I'm afraid...

I'll look into that.

**Comment by clason** [comment]
It's just the `@spell` and `@conceal` captures, I believe -- the script can't find them inside backticks.

**Comment by clason** [comment]
Guess I was wrong; it only reports the first error. It looks like it can't find any captures...

**Comment by ObserverOfTime** [comment]
> Invalid capture @variable.global in highlights for ruby.

I removed this capture since it doesn't actually have any highlighting defined.
I guess the CI doesn't fall back to checking the parent for non-standard captures.

**Comment by clason** [comment]
> I guess the CI doesn't fall back to checking the parent for non-standard captures.

It should; but maybe it's not working properly? @theHamsta

**Comment by ObserverOfTime** [comment]
I'll add a block with non-standard captures for now and see if that appeases the CI.

**Comment by ObserverOfTime** [comment]
Yep, that worked.

**Comment by clason** [comment]
Looks good to me!

**Review by clason — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -89,150 +89,176 @@ effect on highlighting. We will work on improving highlighting in the near futur
...
-@debug
-@error for error `ERROR` nodes.
-@none to disable completely the highlight
-@preproc
-@punctuation.delimiter for `;` `.` `,`
-@punctuation.bracket for `()` or `{}`
-@punctuation.special for symbols with special meaning like `{}` in string interpolation.
+```scheme
+@comment  ; line and block comments
+@debug    ; debugging statements [NOTE: unused]
```
Comment (theHamsta) [inline CONTRIBUTING.md:94]: let's remove this, when nobody can give an example what is meant by debugging statements

**Comment by theHamsta** [comment]
Can we remove all documentation of `TS*` in `doc/`? It would be good to have just one source of truth for the definition of the captures and `TS*` is deprecated anyway. Captures are mostly documented in upstream Neovim.

**Comment by clason** [comment]
> Can we remove all documentation of `TS*` in `doc/`? It would be good to have just one source of truth for the definition of the captures and `TS*` is deprecated anyway. 

Yes, the `TS*` groups should be removed now that Nvim 0.8 is released (see https://github.com/nvim-treesitter/nvim-treesitter/issues/3572)

> Captures are mostly documented in upstream Neovim.

No, they are not. Neovim does and will not _prescribe_ any specific captures (except that any `@foo.bar.<lang>` capture will automatically be mapped to a highlight group of the same name).

**Comment by clason** [comment]
> Yes, the `TS*` groups should be removed now that Nvim 0.8 is released (see #3572)

Done: https://github.com/nvim-treesitter/nvim-treesitter/pull/3656

File: CONTRIBUTING.md (hunk 1 above)
Comment (clason) [reply CONTRIBUTING.md]: @theHamsta is the capture czar, but if it's not used by any current query, I'd suggest leaving it out for now; if somebody ends up needing it, they can easily add it (and provide a good description).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -89,150 +89,176 @@ effect on highlighting. We will work on improving highlighting in the near futur
...
+@text.math             ; math environments (e.g. `$ ... $` in LaTeX)
+@text.environment      ; text environments of markup languages
+@text.environment.name ; text indicating the type of an environment
+@text.reference        ; text references, footnotes, citations, etc.
+
+@text.note             ; info notes
+@text.warning          ; warning notes
+@text.danger           ; danger/error notes
+
+@todo                  ; todo notes [NOTE: unused]
```
Comment (clason) [inline CONTRIBUTING.md:206]: I'd remove that one, too -- it stands out like a sore thumb in this list.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -89,150 +89,176 @@ effect on highlighting. We will work on improving highlighting in the near futur
...
-@preproc
-@punctuation.delimiter for `;` `.` `,`
-@punctuation.bracket for `()` or `{}`
-@punctuation.special for symbols with special meaning like `{}` in string interpolation.
+```scheme
+@comment  ; line and block comments
+@error    ; syntax/parser errors
+@none     ; completely disable the highlight
+@preproc  ; various preprocessor directives & shebangs
+@define   ; preprocessor definition directives [NOTE: unused]
```
Comment (clason) [inline CONTRIBUTING.md:97]: Why did you add it if it's not used? (Honestly curious.)

**Review by clason — COMMENTED** [review]
For the remaining unused captures that match the builtin groups, I'm in principle OK with keeping them (but would drop the "unused" note).

Or better yet, actually use them :) (@gpanders)

**Comment by ObserverOfTime** [comment]
I plan to submit another PR soon that will make some queries more compliant with the documentation and use most if not all of the unused captures. I can drop the note but I'd rather not remove them entirely for the time being.

File: CONTRIBUTING.md (hunk 2 above)
Comment (ObserverOfTime) [reply CONTRIBUTING.md:206]: Maybe we could rename it to `@text.todo`?

File: CONTRIBUTING.md (hunk 3 above)
Comment (ObserverOfTime) [reply CONTRIBUTING.md]: I wanted to document every capture that had a `TS*` equivalent.
I didn't add `@macro` and `@structure` since they didn't have one.

Comment (clason) [reply CONTRIBUTING.md]: Ah, that makes sense. But we don't look back; `TS*` is dead and gone now ;)

File: CONTRIBUTING.md (hunk 2 above)
Comment (clason) [reply CONTRIBUTING.md]: Yes, that would be good (and maybe use it for `@comment` highlight in your follow-up PR).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -89,150 +89,175 @@ effect on highlighting. We will work on improving highlighting in the near futur
...
 
-#### Variables
+```scheme
+@type            ; type or class definitions and annotations
+@type.builtin    ; built-in types
+@type.definition ; type definitions (e.g. `typedef` in C)
+@type.qualifier  ; type qualifiers (e.g. `const`)
 
+@storageclass    ; visibility/life-time/etc. modifiers (e.g. `static`)
+@attribute       ; attribute annotations (e.g. Python decorators)
```
Comment (Diaoul) [inline CONTRIBUTING.md:165]: > I plan to submit another PR soon that will make some queries more compliant with the documentation and use most if not all of the unused captures. I can drop the note but I'd rather not remove them entirely for the time being.

I imagine this is the reason why this is not the case just yet for Python decorators :hugs:
